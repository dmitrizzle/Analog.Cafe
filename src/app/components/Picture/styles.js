// tools
import React from "react"
import Picture from "./components/PictureElement"

// styles
import styled, { css } from "styled-components"
import Color from "color"
import { Caption as PictureCaption, CaptionStyles } from "../CaptionStyles"

// css
// remove `style` prop from Picture HOC
export const Image = styled(({ style, ...props }) => (
  <Picture
    {...props}
    onContextMenu={event => props.protected && event.preventDefault()}
  />
))`
  width: 100%;
  height: auto;
  display: block;
  text-align: center;
  font-style: italic;
  ${props =>
    props.protected &&
    `
		-webkit-touch-callout : none;
		user-select : none;
		pointer-events: none;
    -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
	`};
`

const shadow = css`
  box-shadow: 0 0 ${props => props.theme.size.block.spacing / 2}em
    ${props =>
      Color(props.theme.color.foreground)
        .alpha(props.theme.opacity.least)
        .string()};
`
const bleed = css`
  float: none;
  margin-left: -${props => props.theme.size.block.column.safety}em;
  margin-right: -${props => props.theme.size.block.column.safety}em;
  margin-bottom: 0;
  width: 100vw !important;
  max-width: 100vw !important;
  box-shadow: none;
  border-radius: 0;

  ${props =>
    props.theme.size.breakpoint.min.l`margin-top: ${props =>
      props.theme.size.block.spacing}em;`} ${props =>
  props.feature
    ? props => props.theme.size.breakpoint.min.l`
		margin-left:	calc(( -100vw + ${props =>
      props.theme.size.block.column.maxwidth.m}px )/2 - ${props =>
        props.theme.size.block.column.safety}em );
	`
    : null} ${props =>
  props.feature
    ? props => props.theme.size.breakpoint.min.xxl`
		margin-left:	calc(( -100vw + ${props =>
      props.theme.size.block.column.maxwidth.l}px )/2 - ${props =>
        props.theme.size.block.column.safety}em );
	`
    : null} ${props => props.theme.size.breakpoint.max.m`
		margin-top: 0;
	`};
`
export const Figure = styled.figure`
  overflow: hidden;
  position: relative;
  padding: 0;
  margin: ${props => props.theme.size.block.spacing / 2}em
    ${props => props.theme.size.block.spacing}em
    ${props => props.theme.size.block.spacing}em -${props =>
  props.theme.size.block.column.maxwidth.m / 4}px;
  z-index: ${props => props.theme.layer.up};
  width: 85%;
  float: left;
  background: ${props => props.theme.color.background};
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);

  ${shadow} ${props => props.theme.size.breakpoint.min.xxl`
		width: 				95%;
		margin-left: 	-${props => props.theme.size.block.column.maxwidth.l / 2.75}px;
		margin-right: ${props => props.theme.size.block.spacing}em;
	`} ${props =>
  !props.feature &&
  props.theme.size.breakpoint.max.l`
		//--> Larger figure borders (for figures that aren't featured and are on mobile screens)
		float: none;
		margin: ${props => props.theme.size.block.spacing / 2}em 0 ${props =>
    props.theme.size.block.column.safety}em  -${props =>
    props.theme.size.block.column.safety}em !important;
		width: 75% !important;
		max-width: 66vw !important;
		min-width: ${props => props.theme.size.block.minFigureWIdth}px;

    // this helper graphic hints on tablet-sized devices that no text is going
    // to be float to the right of the image:
    &.focus {
      overflow: visible;
      &::after {
        content: "";
        width: 100vw;
        position: absolute;
        top: 0;
        bottom: 0;
        background: ${props =>
          Color(props.theme.color.foreground)
            .alpha(props.theme.opacity.least / 3)
            .string()};
        z-index: ${props => props.theme.layer.tuck};
      }
    }
	`} ${props =>
  props.feature
    ? bleed
    : props => props.theme.size.breakpoint.max.m`

		margin-left: 0 !important;
		border-radius:	${props => props.theme.effects.borderRadius.small}em;
	`} ${props =>
  props.feature
    ? bleed
    : props => props.theme.size.breakpoint.max.s`
		${bleed}
		//--> Non-featured figures on small screens are not edge-to-edge:
		width: 100% !important;
		max-width: 100vw !important;
		min-width: 0;
		border-radius:	${props => props.theme.effects.borderRadius.small}em;
		${shadow}
	`} &.focus {
    border-top-right-radius: 0;
    border-top-left-radius: 0;
    box-shadow: 0 -${props => props.theme.size.block.border}px 0 ${props =>
  props.theme.color.highlight};
    figcaption {
      box-shadow: 0 ${props => props.theme.size.block.border}px 0
        ${props => props.theme.color.highlight} inset;
    }
    z-index: ${props => props.theme.layer.up + 1};
  }

  textarea {
    ${CaptionStyles};
  }
`

const captionBlock = css`
  ${props => props.theme.size.breakpoint.min.l`
		margin: 0 auto;
		${props => props.theme.size.breakpoint.min.s`
			max-width:	 ${props => props.theme.size.block.column.maxwidth.m}px;
		`}
		${props => props.theme.size.breakpoint.min.xxl`
			max-width:	 ${props => props.theme.size.block.column.maxwidth.l}px;
		`}
	`};
`
export const Caption = styled(PictureCaption)`
  ${props =>
    !props.feature &&
    `
    // for helper graphic that shows grey to the right of figures
    // the background for caption needs to be white
    // and highlight needs to be above
    background: ${props.theme.color.background};
    .focus & {
      box-shadow: 0 ${props.theme.size.block.border}px 0 ${
      props.theme.color.highlight
    } inset;
    }
  `} border-bottom: ${props => props.theme.elements.thickBorder};
  color: ${props =>
    Color(props.theme.color.foreground)
      .alpha(props.theme.opacity.half)
      .string()};
  padding: ${props => props.theme.size.block.column.safety / 2}em
    ${props =>
      props.theme.size.block.column.safety /
      props.theme.size.font.make.smaller}em
    ${props => props.theme.size.block.column.safety * 1.25}em;
  text-align: center;

  max-width: ${props => props.theme.size.block.column.maxwidth.m}px;
  margin: 0 auto;

  div,
  textarea {
    display: inline;
  }
  textarea {
    font-size: 1em !important;
    text-align: center;
    overflow: hidden;
    font-variant: small-caps;
  }
  ${props => props.feature && captionBlock};
`
export const CaptionAuthor = styled.span`
  color: ${props => props.theme.color.foreground};
  display: inline-block;
`
