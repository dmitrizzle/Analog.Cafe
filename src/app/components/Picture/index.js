// tools
import React from "react"
import errorMessages from "../../../constants/messages/errors"

import { ROUTE_AUTHOR_API } from "../../../constants/author"

// components
import { ModalDispatch } from "../../containers/Modal"

// styles
import { Image, Figure, Caption } from "./styles"

// return
export default props => {
  const { src, ...select } = props
  return (
    <Figure {...select}>
      <Image
        {...props}
        protected={
          props.readOnly !== false && process.env.NODE_ENV === "production"
        }
      />
      <figcaption
        style={
          props.nocaption && {
            borderBottom: "8px solid #2c2c2c",
            height: 0,
            overflow: "hidden"
          }
        }
      >
        {props.author ? (
          <Caption>
            {props.children}
            {props.readOnly ? (
              <span style={{ color: "#2c2c2c" }}>
                {" "}
                Image by{" "}
                <span
                  style={props.author.name === "" ? { display: "none" } : null}
                >
                  <ModalDispatch
                    with={
                      props.author.id !== "unknown"
                        ? {
                            request: {
                              url: ROUTE_AUTHOR_API + "/" + props.author.id
                            }
                          }
                        : {
                            info: {
                              title: errorMessages.VIEW_TEMPLATE.PICTURE.title,
                              text: errorMessages.VIEW_TEMPLATE.PICTURE.text,
                              error: props.author.error
                            },
                            id: "errors/author"
                          }
                    }
                  >
                    {props.author.name}
                  </ModalDispatch>.
                </span>
              </span>
            ) : null}
          </Caption>
        ) : (
          <Caption {...props}>
            {props.children}
            {!props.noAuthor &&
              props.readOnly && (
                <span style={{ color: "#2c2c2c" }}> Finding image authorâ€¦</span>
              )}
          </Caption>
        )}
      </figcaption>
    </Figure>
  )
}
