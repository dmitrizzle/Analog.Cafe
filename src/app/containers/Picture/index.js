// tools
import React from "react"
import { getFroth } from "../../../utils/image-froth"
//import localForage from "localforage"

// redux
import { connect } from "react-redux"
import { getInfo } from "../../../actions/pictureActions"

// components
import Picture from "../../components/Picture"
import { PlainTextarea } from "../../components/_controls/InputStyles"
import PictureMenu from "../Composer/containers/ContentEditor/components/PictureMenu"

import { PICTURE_DATA_OBJECT } from "../../../constants/picture"

// export
// let localForageCache
class Figure extends React.PureComponent {
  // state for caption & selection
  constructor(props) {
    super(props)
    this.state = {
      caption: props.node.data.get("caption"),
      src: props.node.data.get("src") || "",
      key: ""
    }
    this.handleChange = this.handleChange.bind(this)
    this.handleTextareaClick = this.handleTextareaClick.bind(this)
    this.handleRemovePicture = this.handleRemovePicture.bind(this)
    this.handleFeaturePicture = this.handleFeaturePicture.bind(this)
  }

  // listeners
  componentWillReceiveProps = nextProps => {
    const caption = nextProps.node.data.get("caption")
    if (caption !== this.state.caption) {
      this.setState({ caption })
    }
  }
  handleChange = event => {
    // format caption text
    let caption = event.target.value
      .replace(/'\b/g, "‘") // opening singles
      .replace(/\b'/g, "’") // closing singles
      .replace(/"\b/g, "“") // opening doubles
      .replace(/\b"/g, "”") // closing doubles
      .replace(/--/g, "—") // em-dashes
      .replace(/\b\.\.\./g, "…") // ellipsis
    const { node, editor } = this.props
    const feature = node.data.get("feature")
    const src = node.data.get("src")
    const key = node.data.get("key") || false
    const file = node.data.get("file") || false
    const properties = { data: { caption, src, feature, key, file } }

    const resolvedState = editor.value
      .change()
      .setNodeByKey(node.key, properties)
    editor.onChange(resolvedState)

    // store DB key if available
    this.setState({ key })
  }
  handleTextareaClick = event => {
    event.preventDefault()
    event.stopPropagation()
  }

  // init
  componentDidMount = () => {
    const { node } = this.props
    const { data } = node
    const caption = data.get("caption")
    const key = data.get("key")
    this.setState({ caption })
    this.loadImage(data.get("file"), key, data.get("src"))

    // store DB key if available
    this.setState({ key })
  }
  loadImage = (file, key, src) => {
    if (!key) {
      this.setState({ src })
      // get image author
      this.props.readOnly && this.props.getInfo(src)
    } else {
      import("localforage").then(localForage => {
        // localForageCache = localForage
        localForage.getItem(key).then(data => {
          const reader = new FileReader()
          reader.addEventListener("load", () =>
            this.setState({ src: reader.result })
          )
          if (
            data &&
            Object.keys(file).length === 0 &&
            file.constructor === Object
          ) {
            reader.readAsDataURL(data)
          } else if (file && file.constructor !== Object) {
            reader.readAsDataURL(file)
          }
        })
      })

      // store DB key if available
      this.setState({ key })
    }
  }

  handleRemovePicture = () => {
    const { node, editor } = this.props
    if (!editor.value.document.getDescendant(node.key)) return
    editor.onChange(editor.value.change().removeNodeByKey(node.key))
  }
  handleFeaturePicture = () => {
    const { node, editor } = this.props
    const previousData = PICTURE_DATA_OBJECT(
      editor.value.document.getChild(node.key).data
    )
    let featureStatus = previousData.feature ? false : true
    editor.onChange(
      editor.value
        .change()
        .setNodeByKey(node.key, {
          type: "image",
          data: { ...previousData, feature: featureStatus }
        })
        .focus()
    )
  }

  render = () => {
    const { attributes, node, isSelected, editor } = this.props
    const { src } = this.state
    const focus = editor.value.isFocused && isSelected
    const className = focus ? "focus" : "nofocus"
    const feature = node.data.get("feature")

    return [
      !this.props.readOnly ? (
        <PictureMenu
          key="PictureMenu"
          focus={focus}
          removePicture={this.handleRemovePicture}
          featurePicture={this.handleFeaturePicture}
        />
      ) : null,
      <Picture
        key="Picture"
        {...attributes}
        readOnly={this.props.readOnly}
        src={src}
        className={className}
        author={
          this.props.pictures[getFroth(src)] &&
          this.props.pictures[getFroth(src)].info.author
        }
        composer={!this.props.readOnly}
        feature={feature}
      >
        {!this.props.readOnly ? (
          <PlainTextarea
            value={this.state.caption}
            placeholder="Add image title, location, camera, film&hellip;"
            onChange={this.handleChange}
            onClick={this.handleTextareaClick}
          />
        ) : (
          <span>{this.state.caption}</span>
        )}
      </Picture>
    ]

    //
    // return src ? <Picture /> : (
    //   <Picture {...attributes} src="" className={className}>
    //     Loading image…
    //   </Picture>
    // )
  }
}

// connect with redux
const mapStateToProps = state => {
  return {
    pictures: state.pictures
  }
}
const mapDispatchToProps = dispatch => {
  return {
    getInfo: src => {
      dispatch(getInfo(src))
    }
  }
}
export default connect(mapStateToProps, mapDispatchToProps)(Figure)
