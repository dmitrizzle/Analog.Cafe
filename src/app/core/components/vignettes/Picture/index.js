import React from "react"
import errorMessages from "../../../../user/constants/errors"

import { ROUTE_AUTHOR_API } from "../../../constants/author"

// components
import { ModalDispatch } from "../../containers/Modal"

// styles
import { Image, Figure, Caption, CaptionAuthor } from "./styles"

// return
export default props => {
  const { src, ...select } = props
  return (
    <Figure {...select}>
      <Image
        {...props}
        protected={
          props.readOnly !== false && process.env.NODE_ENV === "production"
        }
      />
      <figcaption
        style={
          props.nocaption && {
            borderBottom: "8px solid #2c2c2c",
            height: 0,
            overflow: "hidden"
          }
        }
      >
        {props.author ? (
          <Caption>
            {props.children}
            {props.readOnly ? (
              <CaptionAuthor>
                {" "}
                Image by{" "}
                <span
                  style={props.author.name === "" ? { display: "none" } : null}
                >
                  <ModalDispatch
                    with={
                      props.author.id !== "unknown"
                        ? {
                            request: {
                              url: ROUTE_AUTHOR_API + "/" + props.author.id
                            }
                          }
                        : {
                            info: {
                              title: errorMessages.VIEW_TEMPLATE.PICTURE.title,
                              text: errorMessages.VIEW_TEMPLATE.PICTURE.text,
                              error: props.author.error
                            },
                            id: "errors/author"
                          }
                    }
                  >
                    {props.author.name}
                  </ModalDispatch>.
                </span>
              </CaptionAuthor>
            ) : null}
          </Caption>
        ) : (
          <Caption {...props}>
            {props.children}
            {!props.noAuthor &&
              props.readOnly && (
                <CaptionAuthor> Finding image authorâ€¦</CaptionAuthor>
              )}
          </Caption>
        )}
      </figcaption>
    </Figure>
  )
}
