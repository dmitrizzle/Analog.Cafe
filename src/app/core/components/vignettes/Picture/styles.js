import React from "react"
import styled, { css } from "styled-components"

import { styles } from "../../vignettes/Caption"
import Caption from "../Caption"
import Picture from "./components/PictureElement"

// css
// remove `style` prop from Picture HOC
export const Image = styled(({ style, ...props }) => (
  <Picture
    {...props}
    onContextMenu={event => props.protected && event.preventDefault()}
  />
))`
  width: 100%;
  height: auto;
  display: block;
  text-align: center;
  font-style: italic;
  ${props =>
    props.protected &&
    `
		-webkit-touch-callout : none;
		user-select : none;
		pointer-events: none;
    -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
	`};
`

const shadow = css`
  box-shadow: 0 0 ${props => props.theme.size.block.spacing / 2}em
    ${props => props.theme.color.foreground(props.theme.opacity.least)};
`
const bleed = css`
  float: none;
  margin-left: -${props => props.theme.size.block.padding}em;
  margin-right: -${props => props.theme.size.block.padding}em;
  margin-bottom: 0;
  width: 100vw !important;
  max-width: 100vw !important;
  box-shadow: none;
  border-radius: 0;

  ${props =>
    props.theme.size.breakpoint.min.l`margin-top: ${props =>
      props.theme.size.block.spacing}em;`} ${props =>
  props.feature
    ? props => props.theme.size.breakpoint.min.l`
		margin-left:	calc(( -100vw + ${props =>
      props.theme.size.block.column.m}px )/2 - ${props =>
        props.theme.size.block.padding}em );
	`
    : null} ${props =>
  props.feature
    ? props => props.theme.size.breakpoint.min.xxl`
		margin-left:	calc(( -100vw + ${props =>
      props.theme.size.block.column.l}px )/2 - ${props =>
        props.theme.size.block.padding}em );
	`
    : null} ${props => props.theme.size.breakpoint.max.m`
		margin-top: 0;
	`};
`
export const Figure = styled.figure`
  overflow: hidden;
  position: relative;
  padding: 0;
  margin: ${props => props.theme.size.block.spacing / 2}em
    ${props => props.theme.size.block.spacing}em
    ${props => props.theme.size.block.spacing}em -${props =>
  props.theme.size.block.column.m / 4}px;
  z-index: ${props => props.theme.layer.up};
  width: 85%;
  float: left;
  background: ${props => props.theme.color.background()};
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);

  ${shadow} ${props => props.theme.size.breakpoint.min.xxl`
		width: 				95%;
		margin-left: 	-${props => props.theme.size.block.column.l / 2.75}px;
		margin-right: ${props => props.theme.size.block.spacing}em;
	`} ${props =>
  !props.feature &&
  props.theme.size.breakpoint.max.l`
		//--> Larger figure borders (for figures that aren't featured and are on mobile screens)
		float: none;
		margin: ${props => props.theme.size.block.spacing / 2}em 0 ${props =>
    props.theme.size.block.padding}em  -${props =>
    props.theme.size.block.padding}em !important;
		width: 75% !important;
		max-width: 66vw !important;
		min-width: ${props => props.theme.size.block.minFigureWIdth}px;

    // this helper graphic hints on tablet-sized devices that no text is going
    // to be float to the right of the image:
    &.focus {
      overflow: visible;
      &::after {
        content: "";
        width: 100vw;
        position: absolute;
        top: 0;
        bottom: 0;
        background:
            ${props =>
              props.theme.color.foreground(props.theme.opacity.least / 3)};
        z-index: ${props => props.theme.layer.tuck};
      }
    }
	`} ${props =>
  props.feature
    ? bleed
    : props => props.theme.size.breakpoint.max.m`

		margin-left: 0 !important;
		border-radius:	${props => props.theme.effects.borderRadius.small}em;
	`} ${props =>
  props.feature
    ? bleed
    : props => props.theme.size.breakpoint.max.s`
		${bleed}
		//--> Non-featured figures on small screens are not edge-to-edge:
		width: 100% !important;
		max-width: 100vw !important;
		min-width: 0;
		border-radius:	${props => props.theme.effects.borderRadius.small}em;
		${shadow}
	`} &.focus {
    border-top-right-radius: 0;
    border-top-left-radius: 0;
    box-shadow: 0 -${props => props.theme.size.block.border}px 0 ${props =>
  props.theme.color.highlight()};
    figcaption {
      box-shadow: 0 ${props => props.theme.size.block.border}px 0
        ${props => props.theme.color.highlight()} inset;
    }
    z-index: ${props => props.theme.layer.up + 1};
  }

  textarea {
    ${styles};
  }
`

const captionBlock = css`
  ${props => props.theme.size.breakpoint.min.l`
		margin: 0 auto;
		${props => props.theme.size.breakpoint.min.s`
			max-width:	 ${props => props.theme.size.block.column.m}px;
		`}
		${props => props.theme.size.breakpoint.min.xxl`
			max-width:	 ${props => props.theme.size.block.column.l}px;
		`}
	`};
`
export const PictureCaption = styled(Caption)`
  ${props =>
    !props.feature &&
    `
    // for helper graphic that shows grey to the right of figures
    // the background for caption needs to be white
    // and highlight needs to be above
    background: ${props.theme.color.background()};
    .focus & {
      box-shadow: 0 ${
        props.theme.size.block.border
      }px 0 ${props.theme.color.highlight()} inset;
    }
  `} border-bottom: ${props => props.theme.elements.thickBorder};
  color: ${props => props.theme.color.foreground(props.theme.opacity.half)};
  padding: ${props => props.theme.size.block.padding / 2}em
    ${props =>
      props.theme.size.block.padding / props.theme.size.font.make.smaller}em
    ${props => props.theme.size.block.padding * 1.25}em;
  text-align: center;

  max-width: ${props => props.theme.size.block.column.m}px;
  margin: 0 auto;

  div,
  textarea {
    display: inline;
  }
  textarea {
    font-size: 1em !important;
    text-align: center;
    overflow: hidden;
    font-variant: small-caps;
  }
  ${props => props.feature && captionBlock};
`
export const CaptionAuthor = styled.span`
  color: ${props => props.theme.color.foreground()};
  display: inline-block;
`
