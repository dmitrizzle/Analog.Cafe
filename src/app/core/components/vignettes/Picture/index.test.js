import "jest-styled-components"

import { createMockStore } from "redux-test-utils"
import { shallowWithStore } from "enzyme-redux"
import React from "react"

import Picture from "./"

it("Render Picture, matches snapshot", () => {
  const store = createMockStore()
  const element = shallowWithStore(<Picture />, store)
  expect(element).toMatchSnapshot()
})
