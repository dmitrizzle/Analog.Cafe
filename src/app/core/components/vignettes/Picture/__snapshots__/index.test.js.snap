// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Picture, matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(Picture) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "getPictureInfo": [Function],
      "picture": undefined,
      "setModal": [Function],
      "user": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "getPictureInfo": [Function],
        "picture": undefined,
        "setModal": [Function],
        "user": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {
      "store": Object {
        "dispatch": [Function],
        "getAction": [Function],
        "getActions": [Function],
        "getState": [Function],
        "isActionDispatched": [Function],
        "isActionTypeDispatched": [Function],
        "subscribe": [Function],
      },
    },
  },
  Symbol(enzyme.__childContext__): Object {
    "storeSubscription": Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getAction": [Function],
        "getActions": [Function],
        "getState": [Function],
        "isActionDispatched": [Function],
        "isActionTypeDispatched": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": undefined,
    },
  },
}
`;
